easybuild/easyconfigs/e/EDirect/EDirect-19.5.20230508-GCCcore-10.3.0.eb

```diff
@@ -0,0 +1,64 @@
+##
+# This is a contribution from SIB Swiss Institute of Bioinformatics
+# Homepage:     https://www.sib.swiss/research-infrastructure/competence-centers/vital-it
+#
+# Authors::     Sebastien Moretti <sebastien.moretti@sib.swiss>
+#
+##
+easyblock = 'Binary'
+
+name = 'EDirect'
+version = '19.5.20230508'
+
+homepage = 'https://www.ncbi.nlm.nih.gov/books/NBK25501/'
+# See also https://dataguide.nlm.nih.gov/edirect/install.html
+description = """Entrez Direct (EDirect) provides access to the NCBI's suite of
+interconnected databases from a Unix terminal window. Search terms are entered
+as command-line arguments. Individual operations are connected with Unix pipes
+to construct multi-step queries. Selected records can then be retrieved in a
+variety of formats."""
+# software_license = 'Public Domain'
+
+toolchain = {'name': 'GCCcore', 'version': '10.3.0'}
+
+source_urls = ['https://ftp.ncbi.nlm.nih.gov/entrez/entrezdirect/versions/%(version)s/']
+sources = [
+    {'download_filename': 'edirect-%(version)s.tar.gz',
+     'filename': 'edirect-%s.tar.gz' % version,
+     },
+]
+checksums = [
+    '1a73f832a464f150e41c9330e3d8f8f8775f7c39c9352639480289d588fa4889',  # edirect-19.5.20230508.tar.gz
+]
+
+builddependencies = [
+    ('Go', '1.20.4', '', SYSTEM),
+]
+
+dependencies = [
+    ('Python', '3.9.5'),
+    ('Perl',   '5.32.1'),
+]
+
+extract_sources = True
+
+postinstallcmds = [
+    "mkdir %(installdir)s/bin && cd %(installdir)s/cmd/ && "
+    "sed -i 's@target=\"\\$HOME/Misc/scripts/\"@target=\"\\$PWD/../\"@' build.sh && "
+    "./build.sh install && cd ../ && rm -rf cmd/ eutils/ && "
+    "cd bin/ && ln -s ../* . && rm -f easybuild README readme.pdf bin"
```

Comment:
Actually, I think lots of strange things are happening here. 
Is this placing the entire source into the install dir, then running a build in-place then removes things? I think this should just extract the sources into theh builddir instead, then use the installer, which seems to have a target property, and install this into `%(installdir)s/bin` directly instead of PWD.

It also shouldn't be removing the easybuild dir that we put into installation directories (it contains important logs and such).

Either using the GoPackage easyblock instead of their "build.sh" script, or, at least something a long the lines of
```python
easyblock = 'PackedBinary'
...
install_cmd = 'mkdir %(installdir)s/bin && '
install_cmd += "sed -i 's@target=\"\\$HOME/Misc/scripts/\"@target=\"%(installdir)s/bin/@' build.sh && "
install_cmd += './build.sh install'
```

Reply:
The build is done in the install dir, you're right. Yes better to do it in the build dir.

But I think the rest has to be like that:
- `rm -rf cmd/ eutils/` removes the go source files.
- `cd bin/ && ln -s ../* .` scripts and binaries are symlinked in the bin directory
- `rm -f easybuild README readme.pdf bin` **from the bin directory** I then remove symlinks not linked with executables.
So the easybuild/ folder at the root of the install dir is kept.

I will document what is done

Reply:
No? It wouldn't be installed in the source files, so of course no need to do any removal since it was never in the install dir anyway.

And why symlink? Just install it directly in the bin dir, like i suggested instead?
And we should of course not ever remove the easybuildinstallation dir, which won't be in the bin dir anyway.

So i still suggest just doing it the way i propose instead.

Reply:
This package is a tarball with a bunch of bash scripts, and 3 Go binaries to compile.

Once untar-ed and the compilation done the Go sources are not needed any more.

There is no bin directory in the tarball. That's why I created one because I am not sure where the PATH variable points if there is no bin dir. Also not sure if all scripts will be functional if installed not a the root of installdir.

Reply:
In fact without bin directory the install dir itself is in the PATH.

So I compiled in the build dir, simplified the postprocessing step and updated to the last EDirect version.

Reply:
@Micket @verdurin is everything fine now?